tests/partitions_test.py

```diff
@@ -103,6 +106,30 @@ def test_isolate_primary(network, args):
     return network
 
 
+@reqs.description("Isolate and reconnect primary")
+def test_isolate_and_reconnect_primary(network, args):
+    primary, backups = network.find_nodes()
+    with network.partitioner.partition(backups):
+        new_primary, _ = network.wait_for_new_primary(
+            primary, nodes=backups, timeout_multiplier=6
+        )
+        new_tx = reconfiguration.check_can_progress(new_primary)
```

Comment:
I'm not a big fan of test including other tests. If the functionality is generic enough, it should probably be moved to `infra/`? Here, I suggest that we use the `wait_for_commit()` method in `CCFClient` to achieve the same result.

Reply:
`check_can_progress` isn't a test, it's just a utility method, I'll move it to infra (it's also used in rotation and other places). `wait_for_commit()` doesn't submit transactions,